### PR TITLE
Fix: x axis finite labels controls

### LINKED ISSUE #85: Force all x labels
In the case of a bar plot, it is often the case that users want to see all labels rather than get them downsampled. The jump to 0.5 lost that done automatically. Also that jump also lost the ability for force this to happen for any plot. Should be in the axis style.

### PR DESCRIPTION
**Pull request by jonathanrocher**
Bring back text label controls for text x axis. Test by running the new `scripts/explore_dataframe_with_plots.py` and try editing the last bar plot on the first row:
Screen Shot 2020-08-12 at 10 18 26 PM

 
Fixes #85

### CODE REVIEW

File: pybleau/app/plotting/tests/test_plot_styles.py [hunk 1]
```diff
@@ -127,6 +127,12 @@ def test_colored_scatter_plot_style_view(self):
 
 @skipIf(not BACKEND_AVAILABLE, "No UI backend available")
 class TestPlotStyleBehaviors(TestCase):
+    def test_default_plot_style_supports_text_x_axis(self):
+        """ Bar plot styles must support text labels along the x axis. """
+        style = BarPlotStyle()
+        self.assertTrue(style.x_axis_style.support_text_labels)
+        self.assertFalse(style.y_axis_style.support_text_labels)
```
Comment (chamshop) [inline pybleau/app/plotting/tests/test_plot_styles.py:134]: Is it necessary and required that the y axis not support text labels?

File: pybleau/app/plotting/axis_style.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ class AxisStyle(HasStrictTraits):
 
     #: Is there a finite number of labels?
     # (If so, allow forcing the appearance of all of them and label rotation)
-    _finite_labels = Bool
+    support_text_labels = Bool
```
Comment (chamshop) [inline pybleau/app/plotting/axis_style.py:52]: Does the docstring for this need to change, now?

Comment (jonathanrocher) [reply pybleau/app/plotting/axis_style.py:52]: Yes!

File: pybleau/app/plotting/tests/test_plot_styles.py (hunk 1 above)
Comment (jonathanrocher) [reply pybleau/app/plotting/tests/test_plot_styles.py:134]: Not necessary but not supported currently. What plot whould that be? Horizontal bar plot I guess...

**Comment by jonathanrocher** [comment]
Thanks for the review @chamshop !

**Pull request merged**